spring/boot2-actuator-autoconfigure/build.gradle

```diff
@@ -16,8 +16,18 @@ dependencies {
     testImplementation project(':spring:boot2-starter')
 }
 
-// Copy common files from boot2-autoconfigure module to gen-src directory in order to use them as a source set.
+// Copy common files from boot2-autoconfigure module to gen-src directory in order to use them as
+// a source/test set.
+task generateTestSources(type: Copy) {
+    from "${rootProject.projectDir}/spring/boot2-autoconfigure/src/test/java"
+    into "${project.ext.genSrcDir}/test/java"
+    include '**/RetryableArmeriaServerGracefulShutdownLifecycle.java'
+    include '**/RetryableArmeriaServerGracefulShutdownLifecycleConfiguration.java'
+}
+
 task generateSources(type: Copy) {
+    dependsOn(generateTestSources)
```

Comment:
I was a little surprised by this since usually I think the dependency is the other way around  😅 
Having said this, I guess this configuration is used in this project already so SGTM

Reply:
Let me elaborate on `generateSources`. It does not follow Gradle's common dependency convention that the test tasks depend on the source main tasks.
https://github.com/line/armeria/blob/b47aabab483668d28f79b9f72b602f68f445685f/gradle/scripts/lib/java-rpc-proto.gradle#L124-L137
We use a single task name `generateSources` to copy and generate all sources including main and tests for simplicity.

Reply:
I see! Thanks for the explanation 👍